### PR TITLE
An external configuration to override the host ip

### PR DESCRIPTION
**Pull request by DImuthuUpe**
New PR with dependency fix

### CODE REVIEW

File: twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java [hunk 1]
```diff
@@ -189,4 +192,8 @@ public static int getParallelIOAllowance(Config cfg) {
   public static int getRingWorkersPerGroup(Config cfg) {
     return getIntPropertyValue(cfg, RING_GROUPING_WORKER_PER_GROUPS, 128);
   }
+
+  public static String getOverrideHostIP(Config cfg) {
```
Comment (supunkamburugamuve) [inline twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java:196]: We can alternatively put this config to SchedulerContext, that will remove the dependency from comms-api to resource-scheduler

Comment (DImuthuUpe) [reply twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java:196]: @supunkamburugamuve Will move this the resource-scheduler. I thought the network related utilities are going to Communication context

Comment (supunkamburugamuve) [reply twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java:196]: Thank you. The comm package mostly has operator related stuff. That is why I thought this is more suitable in the resource layer.

Comment (DImuthuUpe) [reply twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java:196]: Makes sense

File: twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java [hunk 2]
```diff
@@ -71,6 +71,9 @@
   public static final String RING_GROUPING_WORKER_PER_GROUPS =
       "twister2.network.partition.ring.group.workers";
 
+  public static final String OVERRIDE_HOST_IP =
```
Comment (ahmet-uyar) [inline twister2/api/src/java/edu/iu/dsc/tws/api/comms/CommunicationContext.java:74]: we may rename twister2.network.override.host.ip to twister2.job.master.provided.ip
similarly, OVERRIDE_HOST_IP to JOB_MASTER_PROVIDED_IP

**Comment by DImuthuUpe** [comment]
@supunkamburugamuve @ahmet-uyar I updated the PR as you mentioned

**Comment by supunkamburugamuve** [comment]
@DImuthuUpe thank you.

**Pull request merged**